lte/cloud/go/services/nprobe/nprobe_manager/nprobe_manager.go

```diff
@@ -65,50 +79,97 @@ func getNetworkProbeTasks(networkID string) (map[string]*models.NetworkProbeTask
+// getEvents retrieves all events since start_time from fluentd
+func getEvents(
+	networkID string,
+	state *models.NetworkProbeData,
+	client *elastic.Client,
+) ([]eventdM.Event, error) {
+
+	// build multi-stream es query
+	targetID := state.TargetID
+	startTime := time.Time(state.LastExported).Add(time.Millisecond * 1)
 	queryParams := eventdC.MultiStreamEventQueryParams{
 		NetworkID: networkID,
 		Streams:   nprobe.GetESStreams(),
 		Events:    nprobe.GetESEventTypes(),
 		Tags:      []string{targetID, targetID[4:]},
-		Start:     start_time,
+		Start:     &startTime,
 		Size:      querySize,
 	}
+
 	return eventdC.GetMultiStreamEvents(context.Background(), queryParams, client)
 }
 
+// updateRecordState updates nprobe state with last sequence number and timestamp
+func (np *NProbeManager) updateRecordState(
+	networkID, taskID string,
+	state models.NetworkProbeData,
+	timestamp string,
+	sequenceNumber uint32,
+) error {
+	ptime, err := time.Parse(time.RFC3339, timestamp)
+	if err != nil {
+		return err
+	}
+
+	// update state with last timestamp and sequence nbr
+	state.LastExported = strfmt.DateTime(ptime)
+	state.SequenceNumber = sequenceNumber
+	return np.Storage.StoreNProbeData(networkID, taskID, state)
+}
+
+// processNProbeTask is the main function processing each task, managing state and exporting data
 func (np *NProbeManager) processNProbeTask(networkID string, task *models.NetworkProbeTask) error {
-	// TBD - get the latest state of the task, collect latest events
-	// then process them to create iri records.
-	targetID := task.TaskDetails.TargetID
-	timeSinceReported := time.Time(task.TaskDetails.Timestamp)
-	events, err := getEvents(networkID, targetID, &timeSinceReported, np.ElasticClient)
+	taskID := string(task.TaskID)
+	state, err := np.Storage.GetNProbeData(networkID, taskID)
+	if err != nil {
+		glog.Errorf("Failed to get state for record %s: %v", taskID, err)
+		return err
+	}
+
+	events, err := getEvents(networkID, state, np.ElasticClient)
 	if err != nil {
-		glog.Errorf("Failed to collect events for targetID %s: %s\n", targetID, err)
+		glog.Errorf("Failed to collect events for targetID %s: %s\n", state.TargetID, err)
 		return err
 	}
 
-	// TBD -  retrieve seq nbr from subscriber state and export records
-	var seqNbr uint32 = 0
+	seq := state.SequenceNumber
 	for _, event := range events {
-		_, err := encoding.MakeRecord(&event, task, np.OperatorID, seqNbr)
+		record, err := encoding.MakeRecord(&event, task, np.OperatorID, seq)
 		if err != nil {
-			glog.Errorf("Failed to collect events for targetID %s: %s\n", targetID, err)
+			glog.Errorf("Failed to build record from event %v: %s\n", event, err)
 			continue
 		}
-		seqNbr++
+
+		err = np.Exporter.SendMessageWithRetries(record, np.MaxExportRetries)
+		if err != nil {
+			glog.Errorf("Failed to export record for targetID %s: %s\n", state.TargetID, err)
+			break
```

Comment:
do you want to break if you find one error? No continue? not return?

Reply:
if the export is falling, we want to still update the state with the sequence number of last exported message.